features/step_definitions/skin_steps.rb

```diff
@@ -178,28 +178,36 @@
 
 ### THEN
 
-Then /^the page should have the cached skin "([^"]*)"$/ do |skin_name|
+Then "I should see {string} in the page style" do |css|
+  expect(page).to have_css("style", text: css, visible: false)
+end
+
+Then "I should not see {string} in the page style" do |css|
+  expect(page).not_to have_css("style", text: css, visible: false)
+end
+
+Then "the page should have the cached skin {string}" do |skin_name|
   skin = Skin.find_by(title: skin_name)
-  page.should have_xpath("//link[contains(@href, '#{skin.skin_dirname}')]")
+  expect(page).to have_css("link[href*='#{skin.skin_dirname}']", visible: false)
 end
 
-Then /^the page should not have the cached skin "([^"]*)"$/ do |skin_name|
+Then "the page should not have the cached skin {string}" do |skin_name|
   skin = Skin.find_by(title: skin_name)
-  page.should_not have_xpath("//link[contains(@href, '#{skin.skin_dirname}')]")
+  expect(page).not_to have_css("link[href*='#{skin.skin_dirname}']", visible: false)
```

Comment:
Note that both `.not_to have_css` and `.to have_no_css` RSpec matchers [can wait for AJAX requests](https://github.com/teamcapybara/capybara#asynchronous-javascript-ajax-and-friends); they're functionally equivalent.